pyfstat/make_sfts.py

```diff
@@ -943,6 +941,175 @@ def __init__(
         )
 
 
+class TransientLineWriter(Writer):
+    """Inject a transient line into some data.
+
+    A transient line is defined as a constant amplitude and frequency excess power lasting
+    for a (not necesarilly) finite time span of an observing run.
+
+    In practice, it corresponds to a CW without Doppler or antenna-patern-induced amplitude modulation.
+
+    This uses MFD_v4
+    """
+
+    required_mfd_line_parameters = [
+        "Freq",
+        "phi0",
+        "h0",
+        "cosi",
+        "transientWindowType",
+        "transientStartTime",
+        "transientTau",
+    ]
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        if self.detectors is None:
+            raise ValueError("MakeFakeData_v4 requires detector name to be given")
+        elif len(self.detectors.split(",")) > 1:
+            raise NotImplementedError(
+                "MakeFakeData_v4 does not support more than one detector at a time. "
+                "Please, do it properly"
+            )
+
+    def correct_line_amplitude(self, use_effective_h0, scale_by_duration):
+        """
+        Modify the amplitude parameter h0 to include effects from polarization (cosi)
+        or account for shorter signals.
+
+        use_effective_h0: bool
+            If true, use h0_eff = h0 * sqrt(cosi^4 + 6 * cosi^2 + 1)
+        scale_by_duration: bool
+            If true, scale h0 up by sqrt(t_obs/tau)
+        """
+
+        if use_effective_h0:
+            eta = self.signal_parameters["cosi"]
+            eta2 = eta * eta
+            self.signal_parameters["h0"] *= np.sqrt(eta2 * eta2 + 6 * eta2 + 1)
+        if scale_by_duration:
+            self.signal_parameters["h0"] *= np.sqrt(self.duration / self.transientTau)
+
+    def _parse_args_consistent_with_mfd(self):
+        """
+        Adapt input arguments.
+        Take care of minor inconsistencies between MFD_v4 and MFD_v5
+        """
+        super()._parse_args_consistent_with_mfd()
+
+        if any(
+            key not in self.required_mfd_line_parameters
+            for key in self.signal_parameters
+        ):
+            logging.warning(
+                "Transient lines only require the following parameters:\n"
+                f"{self.required_mfd_line_parameters}.\n"
+                "Any other parameter will be purged from this class now"
+            )
+            params_to_purge = list(
+                set(self.signal_parameters) - set(self.required_mfd_line_parameters)
+            )
+            print("Purging: {}".format(params_to_purge))
```

Comment:
```suggestion
            logging.info("Purging input parameters that are not meaningful for an unmodulated line injection: {}".format(params_to_purge))
```

(black won't like this)